packages/zowe-explorer/src/utils/ProfilesUtils.ts

```diff
@@ -734,15 +734,24 @@ export class ProfilesUtils {
         Gui.infoMessage(responseMsg.join(""), { vsCodeOpts: { modal: true } });
     }
 
-    private static extenderTypeReady = new Map<string, imperative.DeferredPromise<void>>();
+    public static extenderTypeReady: Map<string, imperative.DeferredPromise<void>> = new Map([
+        [
+            "zosmf",
+            ((): imperative.DeferredPromise<void> => {
+                const deferred = new imperative.DeferredPromise<void>();
+                deferred.resolve();
+                return deferred;
+            })(),
+        ],
+    ]);
 
     public static async awaitExtenderType(profileName: string, profInfo: ProfilesCache): Promise<void> {
         const profAttrs = await profInfo.getProfileFromConfig(profileName);
-        if (profAttrs && profAttrs.profType !== "zosmf" && !ProfilesUtils.extenderTypeReady.has(profAttrs.profType)) {
+        if (profAttrs && !ProfilesUtils.extenderTypeReady.has(profAttrs.profType)) {
```

Comment:
Curious why the `profType !== "zosmf"` check was removed in favor of adding a pre-resolved promise for z/OSMF to the map?

Reply:
Just wanted to chime in since I suggested making this change: it removes the need to compare strings and makes the logic identical against all profile types. I'm not biased to either approach, one is being explicit and the other has consistent logic w/o the overhead of a string check 😋